libs/net/traffic_generator.py

```diff
@@ -94,14 +95,19 @@ def _stop_process(vm: BaseVirtualMachine, cmd: str) -> None:
 
 def _is_process_running(vm: BaseVirtualMachine, cmd: str) -> bool:
     try:
-        vm.console(
-            commands=[f"pgrep -fAx '{cmd}'"],
+        for sample in TimeoutSampler(
+            wait_timeout=60,
+            sleep=5,
+            func=vm.console,
+            commands=[f"pgrep -fx '{cmd}'"],
             timeout=_DEFAULT_CMD_TIMEOUT_SEC,
-        )
-        return True
-    except CommandExecFailed as e:
-        LOGGER.info(f"Process is not running on VM {vm.name}. Error: {str(e)}")
+        ):
+            if sample:
+                return True
+    except TimeoutExpiredError as e:
+        LOGGER.warning(f"Process is not running on VM {vm.name}. Error: {str(e.last_exp)}")
         return False
+    return False
```

Comment:
based on slack, please add mypy ignore on the function as the sampler behaves differently and this line will not be reached. ty